### ISSUE TITLE
Custom User-Agent can NOT be set

### ISSUE THREAD

**Issue by PeterDing** [sagebind/isahc#190]
In `isahc = "0.9.2"`, I can not set a custom user-agent to `isahc::HttpClient`. It let every request to always have the default user-agent.

I use `nc -l 8888` to catch the network package of a request which is sent to localhost and then find the issue. The problem is caused by setting default user-agent to `http::Request` before inserting these headers of `HttpClient` to the `http::Request`.

As following code:

1. Setting default user-agent if `request` has not user-agent.
https://github.com/sagebind/isahc/blob/b517b90aad1a3124868c9764d725eaac6110588e/src/client.rs#L791-L794

2. Because above code, `parts.headers` had have the default user-agent !!!.
  So, the costom user-agent in `self.default_headers` can NOT be set to `parts.headers`.
https://github.com/sagebind/isahc/blob/b517b90aad1a3124868c9764d725eaac6110588e/src/client.rs#L984-L990

-------

Following code is my test code.
```rust
use isahc::HttpClient;

let c = HttpClient::builder()
    .default_headers(&[
        ("user-agent", "abc/0.1"),
        ("AAA", "BBB"),
        ("Accept-Encoding", "*/*"),
    ])
    .build()
    .unwrap();

c.get("http://localhost:8888").unwrap();
```

Network package
```
$ nc -l 8888
GET / HTTP/1.1
Host: localhost:8888
Accept: */*
user-agent: curl/7.70.0-DEV isahc/0.9.2
aaa: BBB
accept-encoding: */*
```

**Comment by sagebind** [comment]
Haha, oops! That's no good. Thanks for opening a detailed report, I will look into this as soon as I can.

**Comment by sagebind** [comment]
This is now fixed in 0.9.3!
